designs/ami-selector.md

```diff
@@ -0,0 +1,62 @@
+# Background
+
+The `AMISelector` field of the [`v1alpha1` `AWSNodeTemplate` resource](/pkg/apis/v1alpha1/awsnodetemplate.go)
+is a key-value map with some special key handling (e.g. `aws-ids` is used to pass image IDs).
+
+Most relevant to this design document was the `name` field, which was passed as the `name` filter to
+[`DescribeImages` API](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeImages.html)
+call. This field was removed as a special case in #2978, as there is a risk that without filtering
+by owner, AMI impersonation could happen with various risks (e.g. security vulnerabilities, malware
+such as mining, or cost implications).
+
+While the `Name` tag is still usable, AMI tags are private to an account (so if you want
+to use the AWS bottlerocket AMIs, you'd need to tag them yourself in each account that
+you use)
+
+# Solutions
+
+1. Restore the `name` special case and add a new `owners` special case that is passed
+as the `Owners` argument to `DescribeImages` (this arguments supports both aliases
+such as `self` and `amazon` as well as AWS account IDs). Make `self,amazon` the default
+for this solution.
```

Comment:
> Users that have multiple accounts must now add this to every AMISelector they have to include a list of Account IDs that they trust in order to enable their AMISelector to discover these.

This is true, but this should have always been true if you wish to avoid the security hole that specifying owners fixes. (Also you *can* copy AMIs into an account, you don't necessarily need to trust them - but that's the chore we're trying to avoid by allowing `name` to be specified again - you can just use `tag:Name`, if you first tag them in your own account)